tests/queries/test_qs_combinators.py

```diff
@@ -23,6 +23,15 @@ def test_simple_union(self):
         qs3 = Number.objects.filter(num=5)
         self.assertNumbersEqual(qs1.union(qs2, qs3), [0, 1, 5, 8, 9], ordered=False)
 
+    def test_union_two_evaluations(self):
+        qs = (
+            Number.objects.filter(pk__in=[1, 2])
+            .union(Number.objects.filter(pk__in=[4, 5]))
+            .order_by('num')
+        )
+        self.assertEqual(list(qs.order_by().values_list('pk', flat=True)), [1, 2, 4, 5])
```

Comment:
This is giving `[5, 1, 2, 4]` on PostgreSQL and an empty list on Oracle. It seems the comparison shouldn't care about order if you have `order_by()` which clear's ordering? On the other hand, the assertion on the next line should care about order, right?

Reply:
if order_by() doesn't have any column, order_by is cleared. You are right. I changed it because it's not convenient for our case.